k8s-api/k8s-api.py

```diff
@@ -0,0 +1,147 @@
+from kubernetes import client, config, utils
+from os import path
+
+JOB_NAME = "pi"
+NAMEPACE = "covid"
+YML_JOB_NAME = "k8s-api-job-hello-world"
+
+def create_job_object():
+    # Configureate Pod template container
+    container = client.V1Container(
+        name="pi",
+        image="perl",
+        command=["perl", "-Mbignum=bpi", "-wle", "print bpi(2000)"])
+    # Create and configurate a spec section
+    template = client.V1PodTemplateSpec(
+        metadata=client.V1ObjectMeta(labels={"app": "pi"}),
+        spec=client.V1PodSpec(restart_policy="Never", containers=[container]))
+    # Create the specification of deployment
+    spec = client.V1JobSpec(
+        template=template,
+        backoff_limit=4)
+    # Instantiate the job object
+    job = client.V1Job(
+        api_version="batch/v1",
+        kind="Job",
+        metadata=client.V1ObjectMeta(name=JOB_NAME),
+        spec=spec)
+
+    return job
+
+
+def create_job(api_instance, job):
+    api_response = api_instance.create_namespaced_job(
+        body=job,
+        namespace=NAMEPACE)
+    print("Job created. status='%s'" % str(api_response.status))
+
+
+def update_job(api_instance, job):
+    # Update container image
+    job.spec.template.spec.containers[0].image = "perl"
+    api_response = api_instance.patch_namespaced_job(
+        name=JOB_NAME,
+        namespace=NAMEPACE,
+        body=job)
+    print("Job updated. status='%s'" % str(api_response.status))
+
+
+def delete_job(api_instance,job_name,namespace):
+    api_response = api_instance.delete_namespaced_job(
+        name=job_name,
+        namespace=namespace,
+        body=client.V1DeleteOptions(
+            propagation_policy='Foreground',
+            grace_period_seconds=5))
+    print("Job deleted. status='%s'" % str(api_response.status))
+
+
+def main():
+
+    # Configs can be set in Configuration class directly or using helper
+    # utility. If no argument provided, the config will be loaded from
+    # default location.
+
+    try:
+        config.load_kube_config()
+    except:
+        # load_kube_config throws if there is no config, but does not document what it throws, so I can't rely on any particular type here
+        config.load_incluster_config()
+
+    c = client.Configuration()  # go and get a copy of the default config
+    c.verify_ssl = False  # set verify_ssl to false in that config
```

Comment:
Do we need to do this? I got this warning when tried to list pods: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.43.0.1'. Adding certificate verification is strongly advised.